v3/setup/etcdctl-auth.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+source /etc/environment
+
+etcdctl set /etcdctl/config/root-user root
+etcdctl set /etcdctl/config/read-user read-user
+etcdctl set /etcdctl/config/write-user write-user
+
+ROOT_USERNAME=$(etcdctl get /etcdctl/config/root-user)
+ROOT_PASSWORD=$(etcdctl get /etcdctl/config/root-password)
+READ_USERNAME=$(etcdctl get /etcdctl/config/read-user)
+READ_PASSWORD=$(etcdctl get /etcdctl/config/read-password)
+WRITE_USERNAME=$(etcdctl get /etcdctl/config/write-user)
+WRITE_PASSWORD=$(etcdctl get /etcdctl/config/write-password)
+
+CRED_DIR="/opt/etcdctl"
+if [[ ! -d $CRED_DIR ]]; then
+    sudo mkdir $CRED_DIR -p
+fi
+
+# add root-user
+sudo echo '{"user":"'${ROOT_USERNAME}'", "password":"'${ROOT_PASSWORD}'"}' > $CRED_DIR/root-user.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${ROOT_USERNAME} -XPUT -d "@$CRED_DIR/root-user.json"
+
+# add read-user
+sudo echo '{"user":"'${READ_USERNAME}'", "password":"'${READ_PASSWORD}'"}' > $CRED_DIR/read-user.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${READ_USERNAME} -XPUT -d "@$CRED_DIR/read-user.json"
+
+# add write-user
+sudo echo '{"user":"'${WRITE_USERNAME}'", "password":"'${WRITE_PASSWORD}'"}' > $CRED_DIR/write-user.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${WRITE_USERNAME} -XPUT -d "@$CRED_DIR/write-user.json"
+
+# read access
+etcdctl role add read-only
+etcdctl role grant read-only -path '/*' -read
+
+# write access
+etcdctl role add read-write
+etcdctl role grant read-write -path '/*' -readwrite
+
+# Give read-user read access
+sudo echo '{"user": "'${READ_USERNAME}'", "grant": ["read-only"]}' > $CRED_DIR/read-only.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${READ_USERNAME} -XPUT -d "@$CRED_DIR/read-only.json"
+
+# Give read-write write access
+sudo echo '{"user": "'${WRITE_USERNAME}'", "grant": ["read-write"]}' > $CRED_DIR/read-write.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${WRITE_USERNAME} -XPUT -d "@$CRED_DIR/read-write.json"
```

Comment:
@anshupande I also just realized this but you do what appears to be similar calls for all intents and purposes, have you thought about just wrapping the whole thing in a function and calling that? Are the files in `$CRED_DIR` ever re-used? Or they are loaded and then discarded?

The other remark I will say is maybe you create all of the files up front first and then call storing them later e.g. group all the creation of the files on top and then run your curl calls at the bottom.

```
... define all the files here ...

... run all the curl commands here ...
```

Just my 2 cents